cli/container_prune.go

```diff
@@ -0,0 +1,95 @@
+package main
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/alibaba/pouch/pkg/utils/filters"
+
+	"github.com/spf13/cobra"
+)
+
+// containerPruneDescription is used to remove all stopped containers.
+var containerPruneDescription = "Remove all stopped containers"
+
+// ContainerPruneCommand use to remove all stopped containers.
+type ContainerPruneCommand struct {
+	baseCommand
+
+	force      bool
+	flagFilter []string
+}
+
+// Init initialize "container prune" command.
+func (i *ContainerPruneCommand) Init(c *Cli) {
+	i.cli = c
+	i.cmd = &cobra.Command{
+		Use:   "prune [OPTIONS]",
+		Short: "Remove all stopped containers",
+		Long:  containerPruneDescription,
+		Args:  cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return i.runContainerPrune(args)
+		},
+		Example: i.example(),
+	}
+	i.addFlags()
+}
+
+// addFlags adds flags for specific command.
+func (i *ContainerPruneCommand) addFlags() {
+	flagSet := i.cmd.Flags()
+
+	flagSet.BoolVarP(&i.force, "force", "f", false, "Do not prompt for confirmation")
+	flagSet.StringSliceVarP(&i.flagFilter, "filter", "", nil, "Provide filter values (e.g. 'until=<timestamp>')")
+}
+
+// runContainerPrune is used to remove unused containers.
+func (i *ContainerPruneCommand) runContainerPrune(args []string) error {
+	ctx := context.Background()
+	apiClient := i.cli.Client()
+
+	filter, err := filters.Parse(i.flagFilter)
+	if err != nil {
+		return err
+	}
+
+	if !i.force {
+		fmt.Println("WARNING! This will remove all stopped containers.")
+		fmt.Print("Are you sure you want to continue? [y/N]")
+		var input string
+		fmt.Scanf("%s", &input)
+		if input != "Y" && input != "y" {
+			fmt.Println("Total reclaimed space: 0B")
+			return nil
+		}
+	}
+
+	resp, err := apiClient.ContainerPrune(ctx, filter)
+
+	if err != nil {
+		return err
+	}
+
+	// print resp info
+	if len(resp.ContainersDeleted) > 0 {
+		fmt.Println("Deleted Containers:")
+	}
+	for _, containerDeleted := range resp.ContainersDeleted {
+		fmt.Println(containerDeleted)
+	}
+
+	fmt.Printf("\nTotal reclaimed space: %dB\n", int(resp.SpaceReclaimed))
```

Comment:
Is it better to use this function?　https://github.com/alibaba/pouch/blob/7235f82ae43083197687c714e68bdf1ab425b22c/pkg/utils/utils.go#L27

Reply:
all right